lib/fetch.js

```diff
@@ -125,7 +125,7 @@ var fetchGit = function (state, callback) {
             }
         },
         function (next) {
-            if (isDirectory(tmpPath + '/.git')) {
+            if (isDirectory(tmpPath + '/.git') && !state.options.keepHistory && !(state.flags && state.flags['keep-history'])) {
```

Comment:
Wait a minute, it does have an effect on one of the tests I wrote for it: https://github.com/howardroark/pollinate/blob/develop/tests/test.js#L93